ddev/ddev
# [docs] Refactor install and upgrade instructions

## The Issue

#4722 pointed out how easy it is to miss the upgrading instructions on the [DDEV Installation](https://ddev.readthedocs.io/en/latest/users/install/ddev-installation/) page. We discussed ways of calling attention to a note, which doesn’t address underlying problems:

1. There are lots of ways to install DDEV, which is good for the world and challenging for instructions.
2. Each OS is split into different approaches that have vastly different amounts of explanation.
3. Different install paths reference one another and are not consistent in what they help you accomplish. (Choosing a specific version, for example, or clarifying how to upgrade.)
4. Some details can get lost if you expect to choose a path and not read more broadly. (If I install on macOS with [Homebrew](https://ddev.readthedocs.io/en/latest/users/install/ddev-installation/#homebrew), for example, it can still be easy to miss the Mutagen recommendation because it looks like it’s part of the install script instructions.)

## How This PR Solves The Issue

This PR combines restructuring and edits to solve those problems:

1. **Separates installation and upgrading into separate pages**, allowing each to be more focused:  
    <img src="https://user-images.githubusercontent.com/2488775/228904479-e74bbe26-e4c1-4107-9f75-cc207a6aa4ee.png" width="600" alt="Screenshot of documentation with emphasis on the sidebar, where “DDEV Installation” has a new “Upgrading” page underneath it" />
2. **Prunes wording and removes comments** that are a barrier to the installation task.  
Example: the reader doesn’t benefit from knowing that the install script or Homebrew work on different OS arrangements. They should see clear options and instructions on the tab they’ve chosen without having to read through everything.
3. **Edits each set of instructions (OS + approach) to be self-contained**.  
This introduces some duplicate information for the writer, but it makes it possible to strip out unnecessary explanation and focus the experience for the reader.
4. **Introduces mkdocs-material’s [Collapsible Blocks](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#collapsible-blocks)** to reduce the visual impact of less-common admonitions without removing them:
    <video src="https://user-images.githubusercontent.com/2488775/228668951-b9122b48-ded7-4cac-9ca4-6dc2ddb9571b.mp4" data-canonical-src="https://user-images.githubusercontent.com/2488775/228668951-b9122b48-ded7-4cac-9ca4-6dc2ddb9571b.mp4" controls="controls" muted="muted" class="d-block rounded-bottom-2 border-top width-fit" style="max-height:640px; min-height: 200px" width="600"></video>

Here’s a before and after of the macOS install instructions, which are easier to follow when they’re separated from the upgrading instructions:

![install-before-after](https://user-images.githubusercontent.com/2488775/228937929-31c62f10-08b4-4dbd-9002-711527ce469d.png)

Here’s a comparison of what the reader _has to pay attention to_ in order to update on macOS with Homebrew, before and after the PR:

![upgrade-before-after](https://user-images.githubusercontent.com/2488775/228938145-704040a3-1f43-4976-b628-d2027dea14d9.png)

## Manual Testing Instructions

[Preview changes locally](https://ddev.readthedocs.io/en/stable/developers/testing-docs/#preview-changes) or inspect the [automatic build](https://ddev--4789.org.readthedocs.build/en/4789/users/install/ddev-installation/).

I recommend comparing the proposed [installation page](https://ddev--4789.org.readthedocs.build/en/4789/users/install/ddev-installation/) to the [current one](https://ddev.readthedocs.io/en/latest/users/install/ddev-installation/). Just like the newly-added [upgrading](https://ddev.readthedocs.io/en/latest/users/install/ddev-upgrade/) page, it should feel more focused and easier to follow.

I tested the updated Gitpod install instructions along with the new _updating_ instructions for Gitpod and Codespaces, and everything worked as written.

## Automated Testing Overview

n/a

## Related Issue Link(s)

#4722

## Release/Deployment Notes

~~If it’s possible to redirect URL fragments, I’ll need to do that here.~~ Doesn’t look like it’ll affect any current link fragments, so there shouldn’t be a release impact.


<a href="https://gitpod.io/#https://github.com/ddev/ddev/pull/4789"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
